backend/models/website_app_config.py

```
@@ -0,0 +1,306 @@
+            # Validate URL structure for absolute URLs (supports domains, localhost, IPs)
+            # Check for basic malformed URLs first
+            if '..' in v or v.endswith('://') or '///' in v:
+                raise ValueError('Malformed URL format for favicon')
+                
+            url_pattern = r'^https?://([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?(\.[a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?)*|localhost|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})(:\d{1,5})?(/.*)?$'
+            if not re.match(url_pattern, v):
+                raise ValueError('Invalid URL format for favicon')
+                
+            # Check file extension for absolute URLs
+            if not any(v.lower().endswith(ext) for ext in allowed_extensions):
+                raise ValueError('Favicon URL must end with .ico, .png, or .svg')
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)